components/gfx/display_list/mod.rs

```diff
@@ -786,6 +851,20 @@ pub struct GradientDisplayItem {
     pub stops: Vec<GradientStop>,
 }
 
+impl SizeOf for GradientDisplayItem {
+    fn size_of_excluding_self(&self) -> usize {
+        use libc::c_void;
+        use util::memory::heap_size_of;
+
+        // XXX: we can't measure `stops` via Vec's SizeOf implementation because GradientStop isn't
+        // defined in this module, and we don't want to import GradientStop into util::memory where
+        // the SizeOf trait is defined. So we measure the elements directly.
+        self.base.size_of_excluding_self() +
+            heap_size_of(self.stops.as_ptr() as *const c_void)
```

Comment:
One thing you can do for this kind of thing is to say `struct SizedGradientStops<'a>(&'a GradientStop); impl<'a> SizeOf for SizedGradientStops<'a> { ... }` to get around the coherence limitation. Then you can say `self.base.size_of_excluding_self() + SizedGradientStops(&self.stops).size_of_excluding_self()`. (This is a common Haskell trick; we call it "newtype wrapping".)